src/Dev/Deprecation.php

```diff
@@ -202,40 +216,44 @@ public static function notice($atVersion, $string = '', $scope = Deprecation::SC
         // try block needs to wrap all code in case anything inside the try block
         // calls something else that calls Deprecation::notice()
         try {
-            if (!self::get_is_enabled()) {
-                return;
-            }
-
-            // If you pass #.#, assume #.#.0
-            if (preg_match('/^[0-9]+\.[0-9]+$/', $atVersion ?? '')) {
-                $atVersion .= '.0';
-            }
-
-            // Getting a backtrace is slow, so we only do it if we need it
-            $backtrace = null;
-
-            // Get the calling scope
-            if ($scope == Deprecation::SCOPE_METHOD) {
-                $backtrace = debug_backtrace(0);
-                $caller = self::get_called_method_from_trace($backtrace);
-            } elseif ($scope == Deprecation::SCOPE_CLASS) {
-                $backtrace = debug_backtrace(0);
-                $caller = isset($backtrace[1]['class']) ? $backtrace[1]['class'] : '(unknown)';
+            if ($scope === self::SCOPE_CONFIG) {
+                if (self::get_is_enabled()) {
+                    user_error($string, E_USER_DEPRECATED);
+                } else {
+                    self::$user_error_message_buffer[] = $string;
+                }
             } else {
```

Comment:
I'm not sure this logic is correct ... `self::$user_error_message_buffer[] = $string;` should be called if self::get_is_enabled()` is `false`, not `true` like in the recommendation above

We could switch around the conditional blocks so that instead of

```php
if ($scope === self::SCOPE_CONFIG) {
    if (self::get_is_enabled()) {
```

we have

```php
if (self::get_is_enabled()) {
    if ($scope === self::SCOPE_CONFIG) {
```

but that's going to be just as messy looking and actually read worse IMO (less obvious all the logic for SCOPE_CONFIG is very different because now it's going to be split) so I don't really see the need to refactor this.

Reply:
Sorry, I meant `if (!self::get_is_enabled()) {` for that first condition - that was a bad copy/paste mistake.
I think my recommendation works and is cleaner than the current implementation with this correction. 

If that logic is still incorrect, though, then please point out the error and I'll just accept it from there and we can merge.

Reply:
Logic's wrong, we need the try/catch around everything - self::get_is_enabled() will call Director::isDev() which calls all sorts of other things, including something that currently have Deprecation::notice()

Reply:
Okay. In that case putting the try before the suggested condition would work though.
But ultimately it's a very small point so no point holding back the PR for this. Happy to keep as is.